File: 110-balanced-binary-tree/balanced-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private int Solver(TreeNode root){
        if(root==null) return 0;
        int left = Solver(root.left);
        if(left == -1) return -1;
        int right = Solver(root.right);
        if(right == -1) return -1;
        if(Math.abs(right-left) > 1) return -1;
        return 1 + Math.max(left,right);
    }
    public boolean isBalanced(TreeNode root) {
        return this.Solver(root) != -1;
    }
}

File: 110-balanced-binary-tree/balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        def depth(root):
            if not root:
                return 0, True
            
            left = depth(root.left)
            right = depth(root.right)
            if not(left[1] and right[1]) or abs(right[0]-left[0])>1:
                return 10**9, False

            return (1+max(left[0], right[0]), (left[1] or right[1]))
        _, ans = depth(root)

        return ans